main.tf

```diff
@@ -78,8 +91,14 @@ resource "azurerm_kubernetes_cluster" "aks" {
     tags                  = {}
   }
 
-  identity {
-    type = "SystemAssigned"
+  #  Commented out until we fix loadbalancer public ip provisioning issue, see sp.tf
```

Comment:
I think we just need to assign the role to the Service Principal that is created by the managed identity. You can fetch the MI SP from `azurerm_kubernetes_cluster.aks.identity.principal_id`

Reply:
Cool. I'll try to get this going on a subsequent PR tomorrow to decouple landing a working solution from figuring this out.